server/channels/api4/outgoing_oauth_connection.go

```diff
@@ -354,20 +375,25 @@ func validateOutgoingOAuthConnectionCredentials(c *Context, w http.ResponseWrite
 	// Allow checking connections sent in the body or by id if coming from an already existing
 	// connection url.
 	var inputConnection *model.OutgoingOAuthConnection
-	if c.Params != nil && c.Params.OutgoingOAuthConnectionID != "" {
+
+	bodyReader := io.LimitReader(r.Body, *c.App.Config().ServiceSettings.MaximumPayloadSizeBytes)
+	if err := json.NewDecoder(bodyReader).Decode(&inputConnection); err != nil {
+		c.Err = model.NewAppError(whereOutgoingOAuthConnection, "api.context.outgoing_oauth_connection.validate_connection_credentials.input_error", nil, err.Error(), http.StatusBadRequest)
+		w.WriteHeader(c.Err.StatusCode)
+		return
+	}
+
+	if inputConnection.Id != "" && inputConnection.ClientSecret == "" {
 		var err *model.AppError
-		inputConnection, err = service.GetConnection(c.AppContext, c.Params.OutgoingOAuthConnectionID)
+		var storedConnection *model.OutgoingOAuthConnection
+		storedConnection, err = service.GetConnection(c.AppContext, inputConnection.Id)
 		if err != nil {
 			c.Err = model.NewAppError(whereOutgoingOAuthConnection, "api.context.outgoing_oauth_connection.validate_connection_credentials.app_error", nil, err.Error(), http.StatusInternalServerError)
 			w.WriteHeader(c.Err.StatusCode)
 			return
 		}
-	} else {
-		if err := json.NewDecoder(r.Body).Decode(&inputConnection); err != nil {
-			c.Err = model.NewAppError(whereOutgoingOAuthConnection, "api.context.outgoing_oauth_connection.validate_connection_credentials.input_error", nil, err.Error(), http.StatusBadRequest)
-			w.WriteHeader(c.Err.StatusCode)
-			return
-		}
+
+		inputConnection.ClientSecret = storedConnection.ClientSecret
```

Comment:
In order to test that the connection is valid (and we can retrieve a token) for an already stored connection, the entire connection payload is sent from the frontend but the client secret (since the API does not expose that field), so we need to take the configuration from the frontend and append the client secret to validate the connection.

Reply:
Great, thank you for the explanation.